core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -73,7 +73,10 @@ class SparkEnv (
     val conf: SparkConf) extends Logging {
 
   @volatile private[spark] var isStopped = false
-  private val pythonWorkers = mutable.HashMap[(String, Map[String, String]), PythonWorkerFactory]()
+
+  private case class PythonWorkersKey(
+      pythonExec: String, workerModule: String, daemonModule: String, envVars: Map[String, String])
```

Comment:
Nice! This is so much more readable. Nit: can we add a comment to explain the parameters here (workerModule and daemonModule) briefly?